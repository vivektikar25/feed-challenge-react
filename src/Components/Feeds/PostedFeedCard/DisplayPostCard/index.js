import React from "react";
import PropTypes from "prop-types";
import * as sharedService from "./../../../../SharedService/";
import * as sharedData from "./../../../../SharedData";

import { Card, CardHeader, CardText } from "material-ui/Card";
import CircularProgress from "material-ui/CircularProgress";

const getFormattedDate = date => {
  let formattedDate = sharedService.getFormattedDate(date);
  return formattedDate;
};

const DisplayPostCard = props => {
  return (
    <Card expanded={props.showCardsPostContentFlag}>
      <CardHeader
        style={{ paddingBottom: 5 }}
        title={
          <span>
            <h3 style={{ marginBottom: 0 }}>
              {props.feed.actor} |{" "}
              <span style={{ fontSize: 16, fontWeight: 200 }}>Posted</span>
            </h3>
          </span>
        }
        subtitle={`Published: ${getFormattedDate(props.feed.created_at)}`}
        actAsExpander={true}
        showExpandableButton={false}
      />
      <div
        style={{
          padding: "0px 15px",
          fontSize: "small",
          fontWeight: 100,
          paddingBottom: 10
        }}
      >
        {props.feed.actor} is an English poet, playwright and actor, widely
        regarded as the greatest writer in the English language.
      </div>

      {props.showLoader ? (
        <div
          style={{
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
            paddingBottom: "5%"
          }}
        >
          <CircularProgress />
        </div>
      ) : (
        ""
      )}
      {props.showCardsPostContentFlag ? (
        <CardText expandable={true}>
          <span>Post: {props.cardsPostContent}</span>
          <span
            style={{
              color: "rgba(0, 0, 0, 0.54)",
              display: "block",
              marginBottom: "5px",
              fontSize: 14
            }}
          >
            Updated: {getFormattedDate(props.feed.updated_at)}
          </span>
          <span
            style={{
              color: "#3636d6",
              textDecoration: "underline",
              cursor: "pointer",
              fontSize: "small",
              fontWeight: 100
            }}
          >
            show less
          </span>
        </CardText>
      ) : (
        <div
          style={{
            padding: "0px 15px",
            fontSize: "small",
            fontWeight: 100,
            paddingBottom: 10,
            color: "#3636d6",
            textDecoration: "underline",
            cursor: "pointer"
          }}
        >
          {!props.showLoader ? "show more" : ""}
        </div>
      )}
    </Card>
  );
};

DisplayPostCard.propTypes = {
  feed: PropTypes.shape(sharedData.FeedStruct).isRequired,
  cardsPostContent: PropTypes.string.isRequired,
  showCardsPostContentFlag: PropTypes.bool.isRequired,
  showLoader: PropTypes.bool.isRequired
};
export default DisplayPostCard;
