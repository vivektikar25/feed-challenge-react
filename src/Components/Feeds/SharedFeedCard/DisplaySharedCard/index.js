import React from "react";
import PropTypes from "prop-types";
import * as sharedService from "./../../../../SharedService/";
import * as sharedData from "./../../../../SharedData";

import { Card, CardHeader, CardText } from "material-ui/Card";
import CircularProgress from "material-ui/CircularProgress";

const getFormattedDate = date => {
  let formattedDate = sharedService.getFormattedDate(date);
  return formattedDate;
};

const DisplaySharedCard = props => {
  return (
    <Card expanded={props.showCardsSharedUrlFlag}>
      <CardHeader
        style={{ paddingBottom: 5 }}
        title={
          <span>
            <h3 style={{ marginBottom: 0 }}>
              {props.feed.actor} |{" "}
              <span style={{ fontSize: 16, fontWeight: 200 }}>Shared</span>
            </h3>
          </span>
        }
        subtitle={`Published: ${getFormattedDate(props.feed.created_at)}`}
        actAsExpander={true}
        showExpandableButton={false}
      />
      <div className={"actor-bio"}>
        {props.feed.actor} is an English writer and social critic. He created
        some of the world's best-known fictional characters and is regarded by
        many as the greatest novelist.
      </div>
      {props.showLoader ? (
        <div className={"loader"}>
          <CircularProgress />
        </div>
      ) : (
        ""
      )}
      {props.showCardsSharedUrlFlag ? (
        <CardText expandable={true}>
          <span>
            Shared:{" "}
            <a href={props.cardsSharedUrl} target="_blank">
              {props.cardsSharedUrl}
            </a>
          </span>
          <span className={"feed-updated-date"}>
            Updated: {getFormattedDate(props.feed.updated_at)}
          </span>
          <span className={"feed-card-show-less"}>show less</span>
        </CardText>
      ) : (
        <div className={"feed-card-show-more"}>
          {!props.showLoader ? "show more" : ""}
        </div>
      )}
    </Card>
  );
};

DisplaySharedCard.propTypes = {
  feed: PropTypes.shape(sharedData.FeedStruct).isRequired,
  cardsSharedUrl: PropTypes.string.isRequired,
  showCardsSharedUrlFlag: PropTypes.bool.isRequired,
  showLoader: PropTypes.bool.isRequired
};

export default DisplaySharedCard;
