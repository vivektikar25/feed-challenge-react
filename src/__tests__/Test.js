import React from "react";
import DisplayPostCard from "./../Components/Feeds/PostedFeedCard/DisplayPostCard";
import DisplaySharedCard from "./../Components/Feeds/SharedFeedCard/DisplaySharedCard";

import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

const PostCardProp = {
  id: 1,
  verb: "posted",
  object: "Post:1",
  actor: "Bobby L. Booker",
  created_at: "2018-03-28T16:07:13.840Z",
  updated_at: "2018-03-28T16:07:13.840Z"
};
const ShareCardProp = {
  id: 20,
  verb: "shared",
  object: "Share:10",
  actor: "Yole Fonseca Otero",
  created_at: "2018-03-28T16:07:14.082Z",
  updated_at: "2018-03-28T16:07:14.082Z"
};

// Component Testing //

it("Test post card without post", () => {
  const postedFeedCardRendered = shallow(
    <DisplayPostCard
      feed={PostCardProp}
      cardsPostContent={""}
      showCardsPostContentFlag={false}
      showLoader={false}
    />
  );
  expect(postedFeedCardRendered.contains("Bobby L. Booker")).toEqual(true);
});

it("Test post card with post", () => {
  const postedFeedCardRendered = shallow(
    <DisplayPostCard
      feed={PostCardProp}
      cardsPostContent={"Want to grab pizza"}
      showCardsPostContentFlag={true}
      showLoader={false}
    />
  );
  expect(postedFeedCardRendered.contains("Want to grab pizza")).toEqual(true);
});

it("Test share card without url", () => {
  const sharedFeedCardRendered = shallow(
    <DisplaySharedCard
      feed={ShareCardProp}
      cardsSharedUrl={""}
      showCardsSharedUrlFlag={false}
      showLoader={false}
    />
  );
  expect(sharedFeedCardRendered.contains("Yole Fonseca Otero")).toEqual(true);
});

it("Test share card with url", () => {
  const sharedFeedCardRendered = shallow(
    <DisplaySharedCard
      feed={ShareCardProp}
      cardsSharedUrl={"https://www.flipgive.com/"}
      showCardsSharedUrlFlag={true}
      showLoader={false}
    />
  );
  expect(sharedFeedCardRendered.contains("https://www.flipgive.com/")).toEqual(
    true
  );
});

// Snapshot Testing //

it("renders DisplayPostCard without post", () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <DisplayPostCard
          feed={PostCardProp}
          cardsPostContent={""}
          showCardsPostContentFlag={false}
          showLoader={false}
        />
      </MuiThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders DisplayPostCard with post", () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <DisplayPostCard
          feed={PostCardProp}
          cardsPostContent={"Want to grab pizza"}
          showCardsPostContentFlag={true}
          showLoader={false}
        />
      </MuiThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders DisplaySharedCard without shared url", () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <DisplaySharedCard
          feed={ShareCardProp}
          cardsSharedUrl={""}
          showCardsSharedUrlFlag={false}
          showLoader={false}
        />
      </MuiThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders DisplaySharedCard with shared url", () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <DisplaySharedCard
          feed={ShareCardProp}
          cardsSharedUrl={"https://www.flipgive.com/"}
          showCardsSharedUrlFlag={true}
          showLoader={false}
        />
      </MuiThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
