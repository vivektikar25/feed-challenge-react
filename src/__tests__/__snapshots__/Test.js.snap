// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DisplayPostCard with post 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "boxSizing": "border-box",
          "cursor": "pointer",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "paddingBottom": 5,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          <span>
            <h3
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              Bobby L. Booker
               |
               
              <span
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": 200,
                  }
                }
              >
                Posted
              </span>
            </h3>
          </span>
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          Published: Wednesday, March 28, 2018
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "fontSize": "small",
          "fontWeight": 100,
          "padding": "0px 15px",
          "paddingBottom": 10,
        }
      }
    >
      Bobby L. Booker
       is an English poet, playwright and actor, widely regarded as the greatest writer in the English language.
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      <span>
        Post: 
        Want to grab pizza
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "marginBottom": "5px",
          }
        }
      >
        Updated: 
        Wednesday, March 28, 2018
      </span>
      <span
        style={
          Object {
            "color": "#3636d6",
            "cursor": "pointer",
            "fontSize": "small",
            "fontWeight": 100,
            "textDecoration": "underline",
          }
        }
      >
        show less
      </span>
    </div>
  </div>
</div>
`;

exports[`renders DisplayPostCard without post 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "boxSizing": "border-box",
          "cursor": "pointer",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "paddingBottom": 5,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          <span>
            <h3
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              Bobby L. Booker
               |
               
              <span
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": 200,
                  }
                }
              >
                Posted
              </span>
            </h3>
          </span>
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          Published: Wednesday, March 28, 2018
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "fontSize": "small",
          "fontWeight": 100,
          "padding": "0px 15px",
          "paddingBottom": 10,
        }
      }
    >
      Bobby L. Booker
       is an English poet, playwright and actor, widely regarded as the greatest writer in the English language.
    </div>
    <div
      style={
        Object {
          "color": "#3636d6",
          "cursor": "pointer",
          "fontSize": "small",
          "fontWeight": 100,
          "padding": "0px 15px",
          "paddingBottom": 10,
          "textDecoration": "underline",
        }
      }
    >
      show more
    </div>
  </div>
</div>
`;

exports[`renders DisplaySharedCard with shared url 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "boxSizing": "border-box",
          "cursor": "pointer",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "paddingBottom": 5,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          <span>
            <h3
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              Yole Fonseca Otero
               |
               
              <span
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": 200,
                  }
                }
              >
                Shared
              </span>
            </h3>
          </span>
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          Published: Wednesday, March 28, 2018
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "fontSize": "small",
          "fontWeight": 100,
          "padding": "0px 15px",
          "paddingBottom": 10,
        }
      }
    >
      Yole Fonseca Otero
       is an English writer and social critic. He created some of the world's best-known fictional characters and is regarded by many as the greatest novelist.
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      <span>
        Shared:
         
        <a
          href="https://www.flipgive.com/"
          target="_blank"
        >
          https://www.flipgive.com/
        </a>
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "marginBottom": "5px",
          }
        }
      >
        Updated: 
        Wednesday, March 28, 2018
      </span>
      <span
        style={
          Object {
            "color": "#3636d6",
            "cursor": "pointer",
            "fontSize": "small",
            "fontWeight": 100,
            "textDecoration": "underline",
          }
        }
      >
        show less
      </span>
    </div>
  </div>
</div>
`;

exports[`renders DisplaySharedCard without shared url 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "boxSizing": "border-box",
          "cursor": "pointer",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "paddingBottom": 5,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          <span>
            <h3
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              Yole Fonseca Otero
               |
               
              <span
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": 200,
                  }
                }
              >
                Shared
              </span>
            </h3>
          </span>
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        >
          Published: Wednesday, March 28, 2018
        </span>
      </div>
    </div>
    <div
      style={
        Object {
          "fontSize": "small",
          "fontWeight": 100,
          "padding": "0px 15px",
          "paddingBottom": 10,
        }
      }
    >
      Yole Fonseca Otero
       is an English writer and social critic. He created some of the world's best-known fictional characters and is regarded by many as the greatest novelist.
    </div>
    <div
      style={
        Object {
          "color": "#3636d6",
          "cursor": "pointer",
          "fontSize": "small",
          "fontWeight": 100,
          "padding": "0px 15px",
          "paddingBottom": 10,
          "textDecoration": "underline",
        }
      }
    >
      show more
    </div>
  </div>
</div>
`;
